File: moviebackend/moviebackend/src/main/java/com/example/demo/entities/Theater.java
package com.example.demo.entities;

import jakarta.persistence.*;
import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

@Getter
@Setter
@NoArgsConstructor
@Entity
@Table(name="theater")
public class Theater {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    int theater_id;
    @Column
    String theater_name;
    @Column
    String theater_location;
    @Column
    int total_seats;

    @OneToOne
    @JoinColumn(name = "login_id")
    Login login_id;

    @Column
    int admin_status;
    @Column
    String owner_name;

    @Column
    String licence_no;

    @Column
    String email;
    @Column
    String phone_no;
    @Column
    String address;

    public Theater(String theater_name, String theater_location, int total_seats, Login login_id, String owner_name, String licence_no, String email, String phone_no, String address) {
        this.theater_name = theater_name;
        this.theater_location = theater_location;
        this.total_seats = total_seats;
        this.login_id = login_id;
        this.owner_name = owner_name;
        this.licence_no = licence_no;
        this.email = email;
        this.phone_no = phone_no;
        this.address = address;
    }
}

File: ScheduleBackendDotNet/Models/Theater.cs
using System;
using System.Collections.Generic;

namespace ScheduleBackendDotNet.Models
{
    public partial class Theater
    {
        public Theater()
        {
            Movies = new HashSet<Movie>();
            Screens = new HashSet<Screen>();
            Shows = new HashSet<Show>();
            TheaterSeats = new HashSet<TheaterSeat>();
        }

        public int TheaterId { get; set; }
        public string? TheaterName { get; set; }
        public string? TheaterLocation { get; set; }
        public int? TotalSeats { get; set; }
        public int? LoginId { get; set; }
        public sbyte? TheaterStatus { get; set; }
        public string? OwnerName { get; set; }
        public string? LicenceNo { get; set; }
        public string? Email { get; set; }
        public string? PhoneNo { get; set; }
        public string? Address { get; set; }

        public virtual Login? Login { get; set; }
        public virtual ICollection<Movie> Movies { get; set; }
        public virtual ICollection<Screen> Screens { get; set; }
        public virtual ICollection<Show> Shows { get; set; }
        public virtual ICollection<TheaterSeat> TheaterSeats { get; set; }
    }
}
